src/transactions/TransactionFrame.cpp

```diff
@@ -1772,8 +1792,19 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
             // case
             if (success)
             {
-                app.checkOnOperationApply(op->getOperation(), opResult,
-                                          ltxOp.getDelta());
+                auto delta = ltxOp.getDelta();
+
+                if (protocolVersionIsBefore(ledgerVersion,
+                                            ProtocolVersion::V_8) &&
+                    app.getConfig().EMIT_CLASSIC_EVENTS &&
+                    app.getConfig().BACKFILL_STELLAR_ASSET_EVENTS)
```

Comment:
should call `shouldEmitClassicEvents` here instead of checking the flag combo again.

Reply:
Done